runner/sidecar/sink/kafka/kafka.go

```diff
@@ -41,11 +42,46 @@ func New(ctx context.Context, secretInterface corev1.SecretInterface, x dfv1.Kaf
 	if err != nil {
 		return nil, err
 	}
+
+	producer, err := sarama.NewAsyncProducer(x.Brokers, config)
+	if err != nil {
+		return nil, err
+	}
+
+	logger := sharedutil.NewLogger()
+
 	if x.Async {
-		producer, err := sarama.NewAsyncProducer(x.Brokers, config)
-		if err != nil {
-			return nil, err
-		}
+		config.Producer.Return.Successes = true
+		config.Producer.Return.Errors = true
+
+		// track async success and errors
+		var kafkaMessagesProducedSuccess = promauto.NewCounterVec(prometheus.CounterOpts{
+			Subsystem: "input",
+			Name:      "kafka_produced_success",
+			Help:      "Number of messages successfully produced to Kafka",
+		}, []string{"topic", "type"})
+		var kafkaMessagesProducedErr = promauto.NewCounterVec(prometheus.CounterOpts{
+			Subsystem: "input",
+			Name:      "kafka_produce_errors",
+			Help:      "Number of errors while producing messages to Kafka",
+		}, []string{"topic", "type"})
+
+		// read from Success Channel
+		go func() {
+			// for loop will exit once the producer.Errors() is closed
+			for err := range producer.Errors() {
+				logger.Error(err, "Async to Kafka failed", "topic", err.Msg.Topic)
```

Comment:
I think you should log the source name

Reply:
what do you mean by source name? isn't topic name the source name?

Reply:
well, there's probably no harm in having both